### ISSUE TITLE
action_test broken

### ISSUE THREAD

**Issue by mulkieran** [rhinstaller/blivet#91]
For two reasons:

* When systemMountpoint was introduced to replace _mountpoint, that meant reality was being consulted. Names in action_test are quite realistic, so they can coincide with actual names in /proc/mounts that systemMountpoint consults. This may lead to incorrect status values, with a fmt appearing to be mounted even when not. It may lead to the opposite problem as well, I'm not sure.
* "msdos" is not a recognized format in blivet, and ActionCreateFormat() doesn't think its reasonable to create unknown formats (since tasks branch and since DeviceFormat.formattable is False).

**Comment by dwlehman** [comment]
One option would be to mask the mount cache while running the action tests, eg:

```patch
diff --git a/tests/storagetestcase.py b/tests/storagetestcase.py
index ad1216f..427a19c 100644
--- a/tests/storagetestcase.py
+++ b/tests/storagetestcase.py
@@ -69,6 +69,12 @@ class StorageTestCase(unittest.TestCase):
         self.partition_probe = PartitionDevice.probe
         PartitionDevice.probe = partition_probe
 
+        def nada():
+            pass
+
+        self.get_active_mounts = blivet.formats.fs.mountsCache._getActiveMounts
+        blivet.formats.fs.mountsCache._getActiveMounts = nada
+
     def tearDown(self):
         blivet.devices.StorageDevice.status = self.storage_status
         blivet.devices.DMDevice.status = self.dm_status
@@ -84,6 +90,8 @@ class StorageTestCase(unittest.TestCase):
 
         blivet.devices.PartitionDevice.probe = self.partition_probe
 
+        blivet.formats.fs.mountsCache._getActiveMounts = self.get_active_mounts
+
     def newDevice(self, *args, **kwargs):
         """ Return a new Device instance suitable for testing. """
         device_class = kwargs.pop("device_class")
```

Another option would be to adjust the names of those disks/partitions, eg:

```patch
diff --git a/tests/action_test.py b/tests/action_test.py
index 00d63ad..bafec96 100644
--- a/tests/action_test.py
+++ b/tests/action_test.py
@@ -34,7 +34,7 @@ class DeviceActionTestCase(StorageTestCase):
         """
         super(DeviceActionTestCase, self).setUp()
 
-        for name in ["sda", "sdb", "sdc", "sdd"]:
+        for name in ["mocksda", "mocksdb", "mocksdc", "mocksdd"]:
             disk = self.newDevice(device_class=DiskDevice,
                                   name=name, size=Size("100 GiB"))
             disk.format = self.newFormat("disklabel", path=disk.path,
@@ -42,11 +42,11 @@ class DeviceActionTestCase(StorageTestCase):
             self.storage.devicetree._addDevice(disk)
 
         # create a layout similar to autopart as a starting point
-        sda = self.storage.devicetree.getDeviceByName("sda")
-        sdb = self.storage.devicetree.getDeviceByName("sdb")
+        sda = self.storage.devicetree.getDeviceByName("mocksda")
+        sdb = self.storage.devicetree.getDeviceByName("mocksdb")
 
         sda1 = self.newDevice(device_class=PartitionDevice,
-                              exists=True, name="sda1", parents=[sda],
+                              exists=True, name="mocksda1", parents=[sda],
                               size=Size("500 MiB"))
         sda1.format = self.newFormat("ext4", mountpoint="/boot",
                                      device_instance=sda1,
@@ -54,13 +54,13 @@ class DeviceActionTestCase(StorageTestCase):
         self.storage.devicetree._addDevice(sda1)
 
         sda2 = self.newDevice(device_class=PartitionDevice,
-                              size=Size("99.5 GiB"), name="sda2",
+                              size=Size("99.5 GiB"), name="mocksda2",
                               parents=[sda], exists=True)
         sda2.format = self.newFormat("lvmpv", device=sda2.path, exists=True)
         self.storage.devicetree._addDevice(sda2)
 
         sdb1 = self.newDevice(device_class=PartitionDevice,
-                              size=Size("99.999 GiB"), name="sdb1",
+                              size=Size("99.999 GiB"), name="mocksdb1",
                               parents=[sdb], exists=True)
         sdb1.format = self.newFormat("lvmpv", device=sdb1.path, exists=True)
         self.storage.devicetree._addDevice(sdb1)
```


Which seems preferable?

**Comment by mulkieran** [comment]
It is hard for me to say. You already set the status property of lots of Device* methods to False; perhaps that is another reasonable option.

**Comment by vpodzime** [comment]
I am for the former suggested approach except that it should use ``python-mock`` and Mock instance(s). I've created many unit tests for the oscap-anaconda-addon that use Mock quite heavily and it was really worth it. See for example https://github.com/OpenSCAP/oscap-anaconda-addon/blob/master/tests/common_test.py

**Comment by dwlehman** [comment]
Fixed by 4627e654
